src/components/PostDetails.tsx

```diff
@@ -1,92 +1,54 @@
-import React, { useEffect, useState } from 'react';
+import { useEffect, useState } from 'react';
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
 
 import * as commentsApi from '../api/comments';
-
-import { Post } from '../types/Post';
-import { Comment, CommentData } from '../types/Comment';
-
-type Props = {
-  post: Post;
-};
-
-export const PostDetails: React.FC<Props> = ({ post }) => {
-  const [comments, setComments] = useState<Comment[]>([]);
-  const [loaded, setLoaded] = useState(false);
-  const [hasError, setError] = useState(false);
+import { CommentData } from '../types/Comment';
+import { useAppDispatch, useAppSelector } from '../app/hooks';
+import { actions as commentsActions } from '../features/commentsSlice';
+
+export const PostDetails = () => {
+  const { items, loaded, hasError }
+    = useAppSelector(state => state.comments);
+  const comments = items;
+  const dispatch = useAppDispatch();
   const [visible, setVisible] = useState(false);
 
+  const { selectedPost } = useAppSelector(state => state.selectedPost);
+  const post = selectedPost;
+
   function loadComments() {
-    setLoaded(false);
-    setError(false);
+    dispatch(commentsActions.setLoaded(false));
+    dispatch(commentsActions.setHasError(false));
     setVisible(false);
 
-    commentsApi.getPostComments(post.id)
-      .then(setComments) // save the loaded comments
-      .catch(() => setError(true)) // show an error when something went wrong
-      .finally(() => setLoaded(true)); // hide the spinner
+    commentsApi.getPostComments(post?.id || 0)
+      .then(commentsFromServer => {
+        dispatch(commentsActions.set(commentsFromServer));
+      })
+      .catch(() => dispatch(commentsActions.setHasError(true)))
+      .finally(() => dispatch(commentsActions.setLoaded(true)));
```

Comment:
the same as with `postsSlice` you should create asyncThunk and handle all promise states in slice (in extraReducers)